### ISSUE TITLE
Getfeatureinfo does not show relation reference items for QGIS Server

### ISSUE THREAD

**Issue by NyakudyaA** [qgis/QGIS#36549]
# Problem

I have defined a project with relation reference and serving it through QGIS Server. However, if I try to do a get feature info the relation reference items are not returned.

In QGIS I can see the following:
![relation-reference](https://user-images.githubusercontent.com/2510900/82320586-d0e46680-99d3-11ea-9fd8-83427ef567f5.png)

The WMS layer shows the following
![relation-reference-wms](https://user-images.githubusercontent.com/2510900/82320614-de015580-99d3-11ea-8266-82c9cfc89c2b.png)

Sample project
[bug.zip](https://github.com/qgis/QGIS/files/4650011/bug.zip)

I am running QGIS Server in docker. I tried **QGIS Server 3.10,3.12**

**Comment by elpaso** [comment]
I don't think this has never been implemented. I don't even know if the specs allow for it.
It is indeed possible to return multiple feature collections from a getfeatureinfo response but the "crime" layer is aspatial so it cannot be queried. 
WFS3 will allow you to query an aspatial layer though.

![Screenshot_20200519_192452](https://user-images.githubusercontent.com/142164/82358240-84664e80-9a06-11ea-8dbf-27f52d960e79.png)

**Comment by NyakudyaA** [comment]
> I don't think this has never been implemented. I don't even know if the specs allow for it.
> It is indeed possible to return multiple feature collections from a getfeatureinfo response but the "crime" layer is aspatial so it cannot be queried.
> WFS3 will allow you to query an aspatial layer though.
> 
> ![Screenshot_20200519_192452](https://user-images.githubusercontent.com/142164/82358240-84664e80-9a06-11ea-8dbf-27f52d960e79.png)

@elpaso i can send you another project that has spatial layers with a relationship it still doesn’t work.

**Comment by elpaso** [comment]
No need to. It's not implemented.

**Comment by elpaso** [comment]
@NyakudyaA sorry I was short (on mobile). To elaborate my comment, the actual implementation does not return any related layer. The second part of my comment was more about the possibility/opportunity of implementing this feature and I honestly I can't imagine how it's supposed to work, because AFAIK the GML specification for simple features does not allow to return nested features like in the case we have a
- police_station
     - crime 1
     - crime 2

unless we flattened returning multiple rows for crimes "merged" in a flat view with the police_station.

I mentioned "aspatial" because I image you could retrieve this information in 2 steps:
1. get the station
2. get all crimes for that station

but you cannot query an aspatial layer with WMS, you can with WFS3 though.

Basic answer is probably that we do not support complex features https://docs.geoserver.org/stable/en/user/data/app-schema/complex-features.html#id1

**Comment by andreasneumann** [comment]
Yes, this use case (1:n and n:m) was never implemented in QGIS server, the only thing we have/had for a longer time is to send back the display value instead of the internal value (e.g. in a value relation or relation reference widget send back the value that would be displayed in QGIS desktop, and not the internal value, which might be a number/ID).

I agree though, that it would be very useful to also support 1:n and n:m relations in GeFeatureInfo.

You would have to file an enhancement ticket and we'd need to discuss how to implement it.

**Comment by NyakudyaA** [comment]
I wonder if there is a way to override the current implementation of getfeatureinfo with qgis server plugin that should return all values of a relation. This might be the way to go about this until this is implemented natively in QGIS server.

**Comment by elpaso** [comment]
@NyakudyaA yes, a plugin I/O filter would do just fine, just boring to write because you need to patch XML, unless you are happy with json only, that would be far more palatable.

**Comment by elpaso** [comment]
Ah forgot mention that the big difficulty if you want to follow the specs is IMO how to generate the schema for the getfeatureinfo response, it needs to include the "nested/child" features too.

**Comment by 375000384** [comment]
hi,
I have been using QGIS Server recently, but I don't know how to use WFS3. I have tried many methods, but I couldn't access WFS3. My operating system is Windows 11, and I installed QGIS Server and Apache through OSGeo4W. I wonder if I need to make any additional configurations. I would like to ask for advice from everyone.
